src/main/java/uk/ac/ebi/subs/sheetmapper/SheetMapperService.java

```diff
@@ -0,0 +1,131 @@
+package uk.ac.ebi.subs.sheetmapper;
+
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.web.JsonPath;
+import org.springframework.hateoas.Link;
+import org.springframework.hateoas.UriTemplate;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import uk.ac.ebi.subs.repository.model.sheets.Row;
+import uk.ac.ebi.subs.repository.model.sheets.Sheet;
+import uk.ac.ebi.subs.repository.model.templates.Capture;
+
+import java.net.URI;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Map;
+import java.util.stream.Stream;
+
+@Service
+public class SheetMapperService {
+
+
+    public SheetMapperService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    @Value("${usi.apiRootUrl}")
+    private String rootApiUrl;
+
+    private final RestTemplate restTemplate;
+
+    public void mapSheet(Sheet sheet) {
+        UriTemplate submissionUriTemplate  = new UriTemplate(rootApiUrl + "/submissions/{submissionId}");
+        UriTemplate searchUriTemplate  = new UriTemplate(rootApiUrl + "/{type}/search/by-submissionId-and-alias{?submissionId,alias,projection}");
+        UriTemplate createUriTemplate  = new UriTemplate(rootApiUrl + "/submissions/{submissionId}/contents/{type}");
+
+        String targetType = sheet.getTemplate().getTargetType().toLowerCase();
+        String submissionId = sheet.getSubmission().getId();
```

Comment:
If the sheet is not null, then would it always have a template or submission? Do we need to validate them?

Reply:
I'll add some asserts